packages/noya-state/src/reducers/symbolsReducer.ts

```diff
@@ -167,6 +173,97 @@ export function symbolsReducer(
         });
       });
     }
+    case 'setInstanceSymbolSource': {
+      const [, symbolId] = action;
+
+      const symbolMaster = getSymbols(state).find(
+        (symbol) => symbol.symbolID === symbolId,
+      );
+
+      return produce(state, (draft) => {
+        const symbols = getSelectedLayers(draft) as Sketch.SymbolInstance[];
+
+        if (!symbolMaster) return;
+        symbols.forEach((symbol) => {
+          symbol.frame = {
+            ...symbol.frame,
+            width: symbolMaster.frame.width,
+            height: symbolMaster.frame.height,
+          };
+          symbol.symbolID = symbolId;
+        });
+      });
+    }
+    case 'goToSymbolSource': {
+      const [, symbolId] = action;
+
+      const symbolMaster = getSymbols(state).find(
+        (symbol) => symbol.symbolID === symbolId,
+      );
+
+      return produce(state, (draft) => {
+        if (!symbolMaster) return;
+
+        const page = draft.sketch.pages.find((p) =>
+          p.layers.some((l) => l.do_objectID === symbolMaster.do_objectID),
+        );
+        if (!page) return;
+
+        draft.selectedPage = page.do_objectID;
+        draft.selectedObjects = [symbolMaster.do_objectID];
+      });
+    }
+    case 'setOverrideTextValue':
+    case 'setOverrideTextStyle':
+    case 'setOverrideSymbolId':
+    case 'setOverrideThemeStyle': {
+      const [, name, value] = action;
+
+      return produce(state, (draft) => {
+        const symbols = getSelectedLayers(draft) as Sketch.SymbolInstance[];
+
+        symbols.forEach((symbol) => {
+          const index = symbol.overrideValues.findIndex(
+            (property) => property.overrideName === name,
+          );
+
+          if (action[0] === 'setOverrideSymbolId' && index !== -1) {
+            const symbolMaster = getSymbols(state).find(
+              (symbol) => symbol.symbolID === value,
+            );
+
+            if (!symbolMaster) return;
+            const elements = symbol.overrideValues[index].overrideName.split(
+              '/',
+            );
+            elements.shift();
+
+            symbol.overrideValues[index].overrideName =
+              symbolMaster + '/' + elements.join('/');
+            symbol.overrideValues[index].value = value;
+            return;
+          }
+
+          switch (value) {
+            case '': {
```

Comment:
We should probably pass around `undefined` instead of `''`, even if for the Select element we need `''`

If this only has 2 cases it can be an if/else